### PR TITLE
💅 Add a config for the Patchback GitHub App

### LINKED ISSUE #9385: Write script / GitHub Action for backporting
There is a lot of manual work involved when backporting a PR to a release branch. Ideally, that would only be required if there are conflicts to be solved.

Would be cool to have a script to prepare a backport PR. Then, as a next step, that could be integrated into GitHub Actions so that the "needs backport" label automatically causes a bot to backport the PR to the newest maintenance branch (either when a closed PR is labelled, or when a labelled PR is merged).

Maybe something suitable exists already, I can't imagine us being the only project who needs something like that. I know the Qt Company has something very similar (after I suggested it to them, though I didn't implement it). Theirs is based on their own CI and Gerrit, though.

### LINKED ISSUE #9553: Backport workflow: Extend to allow labeling open PRs
Yep, I was thinking about that as well - that would probably need to be a slightly different trigger (or even a separate workflow) though, which triggers when a PR is merged and then checks that PR's labels. I think ideally we would have both, so that the labels can be applied at any time and the proper thing will happen.

@bluetech is this something you'd be interested in? Personally I should take care of some dayjob stuff at the moment, so pytest 7 keeps me busy enough :slightly_smiling_face:

_Originally posted by @The-Compiler in https://github.com/pytest-dev/pytest/issues/9430#issuecomment-1004813034_

### LINKED ISSUE #9554: Backport workflow: Use access token to run CI on PRs?
Regarding CI not running, right - CI doesn't run when triggered this way - this also happens for the create-pull-request actions and similar. Workaround is to close & reopen, then it triggers. It's also possible to use a user token (pytestbot), but that has its own security considerations.

_Originally posted by @bluetech in https://github.com/pytest-dev/pytest/issues/9430#issuecomment-1004842182_

### LINKED ISSUE #9555: Backport workflow: Add some feedback comments
It would be good to get some (automated) feedback for the backport workflow. The bot could e.g. add a comment with a link to the backported PR if successful (and also take care of adding the "backported" label maybe?). Perhaps more importantly, if not successful, it should add a comment asking for a manual backport.

Right now, it's kind of "fire and forget", and if we don't carefully double-check, we risk accidentally forgetting about backports which should have been done.

### PR DESCRIPTION
**Pull request by webknjaz**
This patch prepares the project's backporting process to start being handled by the Patchback GitHub App [[1]].

Ref #9384
Resolves #9385
Resolves #9553
Resolves #9554
Resolves #9555

[1]: https://github.com/apps/patchback

### CODE REVIEW

**Comment by nicoddemus** [comment]
Thanks @webknjaz, do we have any documentation on how the patchback works?

I assume we just apply the "backport" label at any moment in the PR lifetime, and once it gets merged, it will open a PR backporting the merged change to the correct branch?

**Comment by nicoddemus** [comment]
Should we also remove https://github.com/pytest-dev/pytest/blob/main/.github/workflows/backport.yml?

**Comment by webknjaz** [comment]
> Should we also remove [`main`/.github/workflows/backport.yml](https://github.com/pytest-dev/pytest/blob/main/.github/workflows/backport.yml?rgh-link-date=2024-06-18T15%3A44%3A47Z)?

Not yet. Ronny and I agreed to keep it for a while.

**Comment by webknjaz** [comment]
@nicoddemus

> do we have any documentation on how the patchback works?

Not really, but the source code is pretty clear, I think.
I posted about how it works in https://github.com/pytest-dev/pytest/issues/9554#issuecomment-1284651068 and never got to writing anything more on the topic.
https://github.com/sanitizers/patchback-github-app — there's some feature requests in the issues that I never got to but I think would be useful too.

> I assume we just apply the "backport" label at any moment in the PR lifetime, and once it gets merged, it will open a PR backporting the merged change to the correct branch?

Yep, that's it. It should be a drop-in replacement with no change in how it's used (except for having a few more features than the current workflow).

**Comment by webknjaz** [comment]
For this, install https://github.com/apps/patchback/installations/new/permissions?target_id=8897583&target_type=Organization. You should't need to wait until the PR is merged.

**Review by nicoddemus — APPROVED** [review]
Thanks @webknjaz for the info!

**Comment by bluetech** [comment]
Thanks for fixing this, the old setup was quite annoying :)

**Comment by webknjaz** [comment]
No problem! I would've done this earlier if it was greenlighted..

**Pull request merged**
